internal/graph/check_test.go

```diff
@@ -119,3 +121,200 @@ func TestCheckWithOneConcurrentGoroutineCausesNoDeadlock(t *testing.T) {
 	require.NoError(t, err)
 	require.True(t, resp.Allowed)
 }
+
+func TestCheckDbReads(t *testing.T) {
+	ds := memory.New()
+	defer ds.Close()
+
+	storeID := ulid.Make().String()
+
+	err := ds.Write(context.Background(), storeID, nil, []*openfgav1.TupleKey{
+		tuple.NewTupleKey("document:x", "a", "user:jon"),
+		tuple.NewTupleKey("document:x", "a", "user:maria"),
+		tuple.NewTupleKey("document:x", "b", "user:maria"),
+		tuple.NewTupleKey("document:x", "parent", "org:fga"),
+		tuple.NewTupleKey("org:fga", "member", "user:maria"),
+	})
+	require.NoError(t, err)
+
+	typedefs := parser.MustParse(`
+	type user
+	type org
+      relations
+		define member: [user] as self
+	type document
+	  relations
+		define a: [user] as self
+		define b: [user] as self
+		define union as a or b
```

Comment:
can we have 

```
define union_rewrite: union
```

and then for union we have

```
define union_or_ttu as union or ttu or rewrite_union
```

There should be no changes to the values as they are rewrite.